File: TwoSum.py
def twoSum(nums, target: int):
    hashMap = {}
    if len(nums) == 2:
        return [0, 1]
    for index, item in enumerate(nums):
        hashMap[item] = index
    print(hashMap)
    for index, item in enumerate(nums):
        search = target - item
        if search in hashMap:
            if index != hashMap[search]:
                return [index, hashMap[search]]

twoSum([1,2,3,4,6,5], 10)

File: TwoSum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

vector<int> twoSum(vector<int> nums, int target) {
  unordered_map<int, int> numMap;
  int n = nums.size();

  for (int i = 0; i < n; i++) {
    numMap[nums[i]] = i;
  }

  for (int i = 0; i < n; i++) {
    int complement = target - nums[i];
    if (numMap.count(complement) && numMap[complement] != i) {
      return {i, numMap[complement]};
    }
  }

  return {};
}

int main() { twoSum({5, 5, 5, 5, 5}, 10); }